src/core/lib/iomgr/tcp_client_posix.cc

```diff
@@ -76,6 +76,8 @@ static grpc_error* prepare_socket(const grpc_resolved_address* addr, int fd,
   if (!grpc_is_unix_socket(addr)) {
     err = grpc_set_socket_low_latency(fd, 1);
     if (err != GRPC_ERROR_NONE) goto error;
+    err = grpc_set_socket_tcp_user_timeout(fd, 0 /* set to gRPC default */);
```

Comment:
Can you add support to set this as a channel arg as well ?
Fortunately, the required plumbing is already in place - i.e `channel_args` is passed as an argument to this function and just few lines below , we even have code that parses the args  for `GRPC_ARG_SOCKET_MUTATOR` :)

Reply:
Done